core/src/main/java/io/questdb/cairo/O3OpenColumnJob.java

```diff
@@ -0,0 +1,2825 @@
+                BitmapIndexUtils.valueFileName(pathToPartition.trimTo(pDirNameLen), columnName);
+                dstVFd = openRW(ff, pathToPartition, LOG);
+            }
+
+            if (prefixType != O3_BLOCK_NONE) {
+                partCount++;
+            }
+
+            if (mergeType != O3_BLOCK_NONE) {
+                partCount++;
+            }
+
+            if (suffixType != O3_BLOCK_NONE) {
+                partCount++;
+            }
+        } catch (Throwable e) {
+            LOG.error().$("merge fix error [table=").$(tableWriter.getTableName())
+                    .$(", e=").$(e)
+                    .I$();
+            O3Utils.unmapAndClose(ff, srcDataFixFd, srcDataFixAddr, srcDataFixSize);
+            O3Utils.unmapAndClose(ff, dstFixFd, dstFixAddr, dstFixSize);
+            O3Utils.close(ff, dstKFd);
+            O3Utils.close(ff, dstVFd);
+            tableWriter.o3BumpErrorCount();
+            if (columnCounter.decrementAndGet() == 0) {
+                O3Utils.unmap(ff, srcTimestampAddr, srcTimestampSize);
+                O3Utils.close(ff, srcTimestampFd);
+                Vect.freeMergedIndex(timestampMergeIndexAddr);
+                tableWriter.o3ClockDownPartitionUpdateCount();
+                tableWriter.o3CountDownDoneLatch();
+            }
+            throw e;
+        }
+
+        partCounter.set(partCount);
+        publishMultiCopyTasks(
+                columnCounter,
+                partCounter,
+                columnType,
+                timestampMergeIndexAddr,
+                srcDataFixFd,
+                srcDataFixAddr,
+                srcDataFixOffset,
+                srcDataFixSize,
+                0,
+                0,
+                0,
+                0,
+                srcDataTopOffset,
+                srcDataMax,
+                srcOooFixAddr,
+                srcOooVarAddr,
+                srcOooLo,
+                srcOooHi,
+                srcOooMax,
+                oooPartitionMin,
+                oooPartitionMax,
+                oooPartitionHi,
+                prefixType,
+                prefixLo,
+                prefixHi,
+                mergeType,
+                mergeDataLo,
+                mergeDataHi,
+                mergeOOOLo,
+                mergeOOOHi,
+                suffixType,
+                suffixLo,
+                suffixHi,
+                dstFixFd,
+                dstFixAddr,
+                dstFixSize,
+                0,
+                0,
+                0,
+                dstFixAppendOffset1,
+                dstFixAppendOffset2,
+                0,
+                0,
+                dstKFd,
+                dstVFd,
+                indexBlockCapacity,
+                srcTimestampFd,
+                srcTimestampAddr,
+                srcTimestampSize,
+                tableWriter,
+                indexWriter
+        );
+    }
+
+    private static void mergeVarColumn(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 106 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 2155-2208 into a separate method.